### ISSUE TITLE
EC2 auto-discovery issues

### ISSUE THREAD

**Issue by r0mant** [gravitational/teleport#17359]
A few issues have come up while testing new EC2 auto-discovery.

### Default install script doesn't work on Ubuntu

It fails with `jq: not found` error which hints that `apt-get install teleport jq` doesn't run. The suspicion is that it doesn't run because `-y` flag is not used so none of the packages are installed because it runs in interactive mode and requires user confirmation?

Once fixed, let's verify on:

- [x] Ubuntu 18.04
- [x] Ubuntu 20.04
- [x] Ubuntu 22.04

### Hard to debug SSM failures

When SSM command fails, you have to go to AWS console, find it and see its output. Command output should be available in the `GetCommandInvocationOutput`, let's make sure we include it in the audit event.

Check if the included command output is truncated in the AWS response and if not, let's make sure to truncate it to some reasonable size (e.g. 10kb) to avoid large events. Truncate from the head, not tail, so if the script fails the errors are included.

### Misleading exit code in SSM events

Exit code is currently `0` when status is `Failed` which is confusing.

```yaml
{
  "account_id": "165258854585",
  "cluster_name": "tele1c",
  "code": "TDS00W",
  "command_id": "f38d983b-5b61-46fa-a4a5-2f5f5b9153b1",
  "ei": 0,
  "event": "ssm.run",
  "exit_code": 0,
  "instance_id": "i-0bfde635b9df7f8d7",
  "region": "us-east-1",
  "status": "Failed",
  "time": "2022-10-12T22:03:41.688Z",
  "uid": "6f3ded82-5173-430b-be7a-2907287ac9bb"
}
```

### SSM events are displayed as "unknown" in audit log in web UI

We did add them to web assets:

https://github.com/gravitational/webapps/pull/1192
https://github.com/gravitational/webapps/pull/1223

But @stevenGravy says he's still seeing them as "unknown" in the web UI. Let's verify on our side on latest v10, v11 and master.

**Comment by stevenGravy** [comment]
@lxea this is resolved in `10.3.2`.  It was missing in `10.3.1` which is the version I was running.

**Comment by zmb3** [comment]
@r0mant @lxea is this resolved now?

**Comment by matt-moeller** [comment]
I'm still experiencing this issue, ubuntu 16.04 LTS, below is the error from SSM. We have a fair amount of these in our fleet, is there a known workaround?

```% Total % Received % Xferd Average Speed Time Time Time Current

Dload Upload Total Spent Left Speed

0 0 0 0 0 0 0 0 --:--:-- --:--:-- --:--:-- 0 100 3163 100 3163 0 0 31067 0 --:--:-- --:--:-- --:--:-- 31316

W: GPG error: https://apt.releases.teleport.dev/ubuntu xenial InRelease: The following signatures couldn't be verified because the public key is not available: NO_PUBKEY C87ED53A6282C411

W: The repository 'https://apt.releases.teleport.dev/ubuntu xenial InRelease' is not signed.

E: There were unauthenticated packages and -y was used without --allow-unauthenticated

/tmp/installTeleport.sh: 30: /tmp/installTeleport.sh: jq: not found

/tmp/installTeleport.sh: 31: /tmp/installTeleport.sh: jq: not found

sudo: /usr/local/bin/teleport: command not found

Failed to execute operation: No such file or directory

failed to run commands: exit status 1```
